src/Confluent.Kafka/Serialization/LongDeserializer.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+
+using System;
+
+namespace Confluent.Kafka.Serialization
+{
+    /// <summary>
+    /// Derializer for Int64 data
+    /// </summary>
+    public class LongDeserializer : IDeserializer<long>
+    {
+        /// <summary>
+        /// Deserialize data to Int64
+        /// </summary>
+        /// <param name="data">input data</param>
+        /// <returns>deserialize result</returns>
+        public long Deserialize(byte[] data)
+        {
+            if (data == null)
+            {
+                // in accordance with java client.
+                return default(long);
```

Comment:
Since tombstones for compacted topics is an integral part of Kafka I think all serializers should allow null values and all deserializers to support passing null, without having to have two different variants of everything.

Having just Object will loose type-safe, right?

So I guess that leaves us with Nullable